### ISSUE TITLE
bug: Cannot use history.pushState() / replaceState() with custom scheme on Android

### ISSUE THREAD

**Issue by root14** [ionic-team/capacitor#6981]
I dont think this is related to capacitor but i open this ticket to  have your opinions about this.

Few days ago i noticed i cant navigate anymore in my capacitor application on Android with a cusmos scheme. After some investigation i found a new DOM exception (Exception from code reproduction on a fresh repo provided below):
```
SecurityError: Failed to execute 'replaceState' on 'History': A history state object with URL 'iamcustom://localhost/page1' cannot be created in a document with origin 'iamcustom://' and URL 'iamcustom://localhost/'.
Error: Failed to execute 'replaceState' on 'History': A history state object with URL 'iamcustom://localhost/page1' cannot be created in a document with origin 'iamcustom://' and URL 'iamcustom://localhost/'.
```

### Capacitor Version

```
 Capacitor Doctor

Latest Dependencies:

  @capacitor/cli: 5.4.2
  @capacitor/core: 5.4.2
  @capacitor/android: 5.4.2
  @capacitor/ios: 5.4.2

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 5.4.2
  @capacitor/core: 5.4.2
  @capacitor/android: 5.4.2

[success] Android looking great! 👌
```

### Platform(s)
Android

### Current Behavior
When using custom scheme for Android in capacitor.config.ts, application is not able to navigate using pushState / replaceState.

### Code Reproduction
I made a repro repo, a fresh capacitor install with an Angular app. The app have 2 pages, with a router that redirect to /page1. Page1 never load and the app stay blank because of the DOM exception (Angular use pushState / replaceState to navigate).

https://github.com/root14/capacitor-replace-state-bug-reproduction

### Other Technical Details

`npm --version` output: 9.5.1

`node --version` output: v18.16.0

### Additional Context

If i go back to http scheme, everything work. 

I found some commit in chromium project that aimed to be spec compliant with: 
https://html.spec.whatwg.org/multipage/nav-history-apis.html#can-have-its-url-rewritten

Commit like: 
https://chromium-review.googlesource.com/c/chromium/src/+/4701743

Am i wrong to think that it is an  os / browser update on my devices that broke the navigation with pushState / replaceState + a custom scheme ?

**Comment by jcesarmobile** [comment]
Yeah, chromium "broke" something on version 117, we are aware of it, but sadly there is nothing we can do about it.

Try reporting to chromium, but they don't recommend using custom schemes, so they will probably close the issue and tell you to not use custom schemes as they have done with other issues related to using custom schemes.

**Comment by jcesarmobile** [comment]
we have published a blog entry about this
https://ionic.io/blog/capacitor-android-customscheme-issue-with-chrome-117
